dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.tracing
+
+import datadog.trace.api.DDTags
+import datadog.trace.api.interceptor.MutableSpan
+import io.opentracing.Span
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format.Builtin.TEXT_MAP_INJECT
+import io.opentracing.propagation.TextMapInject
+import io.opentracing.tag.Tags
+import io.opentracing.util.GlobalTracer
+import java.io.PrintWriter
+import java.io.StringWriter
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+
+/**
+ * Provides automatic trace integration for [OkHttpClient] by way of the [Interceptor] system.
+ *
+ * This interceptor will create a [Span] around the request and fill the request information
+ * (url, method, status code, optional error). It will also propagate the span and trace
+ * information in the request header to link it with backend spans.
+ *
+ * If you use multiple Interceptors, make sure that this one is called first.
+ *
+ * To use:
+ * ```
+ *   OkHttpClient client = new OkHttpClient.Builder()
+ *       .addInterceptor(new TracingInterceptor())
+ *       .build();
+ * ```
+ */
+class TracingInterceptor : Interceptor {
+
+    // region Interceptor
+
+    /** @inheritdoc */
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val tracer = GlobalTracer.get()
+        // val activeSpan = tracer.activeSpan() ?: TracesFeature.currentActivitySpan()
```

Comment:
From all my test, this `tracer.activeSpan()` always returns null because there's no span on the thread where the request is processed. 
But then again it's not an issue I think that the okhttp request doesn't have a parent span (for now), caus we don't know what the "logical" parent should be (could be the current activity, or a background service or something else entirely.

Reply:
It is normal to return null....because like you said it is a fresh new Thread and it doesn't have an active span yet. What would be interesting to test is that if you start to okHttp requests one after the other if the Thread used will be the same or if it's using some sort of a Thread Pool. I guess you should remove that commented code then and keep and only keep the part with the tracer.activeSpan(). This might be useful if let's say for example if by any chance 2 calls are sharing the same thread ...not sure though this is a realistic scenario.

Reply:
comment removed